config/routes.rb

```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
   get "/pages/:page", to: "pages#show"
+  get "/supplier_dashboard", to: "suppliers#dashboard"
```

Comment:
This looks like a singleton resource rather than a static page to me. https://guides.rubyonrails.org/routing.html#singular-resources

 If you do resource : supplier_dashboard, only: :index  you can get rid of this unrestful route.

Also, the dashboard would fetch all sorts of data and I don't think you want that in the suppliers controller. A SupplierDashboardController would probably be a more suitable place for this. The index action would just conventionally render the right view rather than having to do render template: "pages/supplier_dashboard"

Reply:
I'm not sure which of `index` vs `show` to use. 

I think `index` is meant to show a list of resources, which is probably not what we are doing here. But admittedly I didn't use singular resources much, so I'm not too familiar with the standard conventions.